solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java

```diff
@@ -539,7 +616,115 @@ private  SimpleOrderedMap getRangeCountsIndexed() throws IOException {
 
   private Query[] filters;
   private DocSet[] intersections;
+  
   private void rangeStats(Range range, int slot, boolean hasSubFacets) throws IOException {
+    Query rangeQ = getRangeQuery(range);
+    // TODO: specialize count only
+    DocSet intersection = fcontext.searcher.getDocSet(rangeQ, fcontext.base);
+    if (hasSubFacets) {
+      filters[slot] = rangeQ;
+      intersections[slot] = intersection;  // save for later  // TODO: only save if number of slots is small enough?
+    }
+    int num = collect(intersection, slot, slotNum -> { return new SlotAcc.SlotContext(rangeQ); });
+    countAcc.incrementCount(slot, num); // TODO: roll this into collect()
+  }
+
+  private void dvRangeStats(List<Range> rangeList, List<Range> otherList, boolean hasSubFacets) throws IOException {
+    List<Range> allRanges = new ArrayList<>(rangeList.size() + otherList.size());
+    allRanges.addAll(rangeList); // first rangeList items are sorted by low
+    allRanges.addAll(otherList);
+    
+    final FieldType ft = sf.getType();
+    final String fieldName = sf.getName();
+    final NumberType numericType = ft.getNumberType();
+    if (numericType == null) {
+      throw new IllegalStateException();
+    }
+
+    final Iterator<LeafReaderContext> ctxIt = fcontext.searcher.getIndexReader().leaves().iterator();
+    LeafReaderContext ctx = null;
+    NumericDocValues longs = null;
+    IntHashSet[] sets = hasSubFacets ? new IntHashSet[intersections.length] : null;
+    
+    for (DocIterator docsIt = fcontext.base.iterator(); docsIt.hasNext(); ) {
```

Comment:
Most of this was ripped, in some capacity, from the params based faceting code flow.